org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java

```diff
@@ -810,7 +810,7 @@ public void checkTerminalFragmentCalledFromTerminalRule(final RuleCall call) {
 							call, 
 							XtextPackage.Literals.RULE_CALL__RULE,
 							ValidationMessageAcceptor.INSIGNIFICANT_INDEX,
-							null);
+							ILLEGAL_RULE_REFERENCE);
```

Comment:
There are other validations, that appear to do similar validation, e.g. in line 796. Please use a more specific field name or provide an issue code for the other cases, too.